File: HSGoogleDrivePicker-Demo/HSGoogleDrivePicker-Demo/ViewController.m
//
//  ViewController.m
//  HSGoogleDrivePicker-Demo
//
//  Created by Rob Jonson on 14/08/2017.
//  Copyright © 2017 HobbyistSoftware. All rights reserved.
//

#import "ViewController.h"
#import "HSDrivePicker.h"

//REQUIRED STEPS TO RUN DEMO
// 1) Provide your secret below
// 2) Add your 'GoogleService-Info.plist' plist to the project
// 3) Configure the correct callback URL
//
// See documentation for full details: https://github.com/ConfusedVorlon/HSGoogleDrivePicker#getting-your-api-keys



@interface ViewController ()

@property (weak, nonatomic) IBOutlet UILabel *pickedFile;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)pickFile:(id)sender {
    
    HSDrivePicker *picker=[[HSDrivePicker alloc] initWithSecret:@"YOUR SECRET HERE"];
    
    
    [picker pickFromViewController:self
                    withCompletion:^(HSDriveManager *manager, GTLDriveFile *file) {
                        //update the label
                        self.pickedFile.text = [NSString stringWithFormat: @"selected: %@",file.name];
                        
                        //Download the file
                        NSString *destinationPath = [NSTemporaryDirectory() stringByAppendingPathComponent:file.name];
                        [manager downloadFile:file toPath:destinationPath withCompletionHandler:^(NSError *error) {
                            
                            if (error)
                            {
                                NSLog(@"Error downloading : %@", error.localizedDescription);
                            }
                            else
                            {
                                NSLog(@"Success downloading to : %@", destinationPath);
                            }
                            
                        }];
                    }];
}





@end

File: HSGoogleDrivePicker-Demo/HSGoogleDrivePicker-Demo/ViewController.swift
//
//  ViewController.m
//  HSGoogleDrivePicker-Demo
//
//  Created by Rob Jonson on 14/08/2017.
//  Copyright © 2017 HobbyistSoftware. All rights reserved.
//

import HSGoogleDrivePicker
import UIKit

class ViewController: UIViewController, UINavigationBarDelegate {

    
    @IBOutlet weak var pickedFile: UILabel!
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    func position(for bar: UIBarPositioning) -> UIBarPosition {
        return .topAttached
    }
    
    @IBAction func pickFile(_ sender: Any) {
        
        let picker = HSDrivePicker(secret: "YOUR SECRET HERE")
        
        
        picker?.pick(from: self, withCompletion: { manager, file in
            //update the label
            if let name = file?.name {
                self.pickedFile.text = "selected: \(name)"
            }
            
            //Download the file
            guard let fileName = file?.name else {
                print("Error: File has no name")
                return
            }
            
            let destURL =  URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(fileName)
            let destinationPath = destURL.absoluteString
            manager?.downloadFile(file, toPath: destinationPath, withCompletionHandler: { error in
                
                if error != nil {
                    print("Error downloading : \(error?.localizedDescription ?? "")")
                } else {
                    print("Success downloading to : \(destinationPath)")
                }
                
            })
        })
    }
}
